examples/step-90/step-90.cc

```diff
@@ -0,0 +1,1344 @@
+                                 TraceFEM_ScratchData<dim> &scratch_data,
+                                 TraceFEM_CopyData<dim>    &copy_data) {
+      // This cell worker does not do anything for non-intersected cells
+      if (mesh_classifier.location_to_level_set(cell) ==
+            NonMatching::LocationToLevelSet::intersected &&
+          cell->is_locally_owned())
+        {
+          // Once we now that the cell is intersected, we construct the unfitted
+          // quadratures for the solutions FE space on the cell.
+          scratch_data.non_matching_fe_values.reinit(cell);
+          copy_data.reinit(cell,
+                           scratch_data.fe_values.get_fe().n_dofs_per_cell());
+          copy_data.cell_matrix = 0;
+          copy_data.cell_rhs    = 0;
+          const std::optional<NonMatching::FEImmersedSurfaceValues<dim>>
+            &surface_fe_values =
+              scratch_data.non_matching_fe_values.get_surface_fe_values();
+          const std::vector<double> &surfJxW =
+            surface_fe_values->get_JxW_values();
+
+          // The accumulation of the surface integrals, including the forcing,
+          // is performed here.
+          for (unsigned int q : surface_fe_values->quadrature_point_indices())
+            {
+              const Tensor<1, dim> &normal =
+                surface_fe_values->normal_vector(q);
+
+              for (const unsigned int i : surface_fe_values->dof_indices())
+                {
+                  copy_data.cell_rhs(i) +=
+                    right_hand_side.value(
+                      surface_fe_values->quadrature_point(q)) *
+                    surface_fe_values->shape_value(i, q) * surfJxW[q];
+
+                  for (const unsigned int j : surface_fe_values->dof_indices())
+                    {
+                      // This is the local contribution to the global linear
+                      // system
+                      copy_data.cell_matrix(i, j) +=
+                        1.0 *
+                        (surface_fe_values->shape_value(i, q) *
+                         surface_fe_values->shape_value(j, q)) *
+                        surfJxW[q];
+                      copy_data.cell_matrix(i, j) +=
+                        (1.0 *
+                         (surface_fe_values->shape_grad(i, q) -
+                          (normal * surface_fe_values->shape_grad(i, q)) *
+                            normal) *
+                         (surface_fe_values->shape_grad(j, q) -
+                          (normal * surface_fe_values->shape_grad(j, q)) *
+                            normal)) *
+                        surfJxW[q];
+                    }
+                }
+            }
+
+          // The normal-gradient volume stabilization form needs a bulk cell
+          // integration while other types of stabilization may need face
+          // quadratures, for example. So we check it first.
+          if (stabilization_scheme.needs_cell_worker())
```

Comment:
The name is chosen unfortunately!? I guess it is `needs_stabilization()`!? Also, I am not sure if we need a separate class for the stabilization. You only have a single implementation... Let's keep the cope simple and  inline the functions here.